clang/lib/Driver/Driver.cpp

```diff
@@ -4920,7 +4925,11 @@ class OffloadingActionBuilder final {
           return ABRT_Success;
 
         Action *DeviceCompilerInput = nullptr;
-        for (Action *&A : SYCLDeviceActions) {
+        const DeviceTargetInfo &DevTarget = SYCLTargetInfoList.back();
+        for (auto TargetActionInfo :
+             llvm::zip(SYCLDeviceActions, SYCLTargetInfoList)) {
+          Action *&A = std::get<0>(TargetActionInfo);
```

Comment:
How to interpret this variable? Reference to a pointer of type Action? I'm seeing this type of var for the first time :) What does it do?

Reply:
That's basically what it is - it's a reference to the pointer, modifying the pointer when updated below.